Update LLVM development policy

MForster opened the pull request:
State explicitly that we want to eliminate all differences between
`apple/master` and `llvm.org/master`. All new LLVM development should happen
upstream.

MForster committed:
Update LLVM development policy

State explicitly that we want to eliminate all differences between
`apple/master` and `llvm.org/master`. All new LLVM development should happen
upstream.

rjmccall reviewed (commented):
This should cover both `apple/master` and `swift/master`.

rjmccall commented:
I think Ted should sign off on the final wording here.

MForster commented:
> This should cover both `apple/master` and `swift/master`.

I think we said that for now the `swift/master` rules don't change. Are you asking to be explicit about that or are you suggesting to tighten those policies as well?

rjmccall commented:
> > This should cover both `apple/master` and `swift/master`.
> 
> I think we said that for now the `swift/master` rules don't change. Are you asking to be explicit about that or are you suggesting to tighten those policies as well?

I'd just forgotten that.  We should be explicit, though.

MForster committed:
Submitting to swift/master-next is still allowed.

Add additional clarification that submitting to `swift/master-next` continues to be allowed.

MForster commented:
> We should be explicit, though.

I've added a note to that effect.

MForster commented:
@swift-ci Please test

tkremenek commented:
This looks great to me.
